File: trunk/tests/java/PreAndOld.java


//@+ CheckArithOverflow = no

//@ logic integer f{L}(integer n) = n+PreAndOld.y;


class PreAndOld {

    static int y;
    
    /*@ ensures \result == \old(f(x))
      @      && \result == f{Old}(x)
      @      && \result == \at(f(x),Pre);
      @*/
    int g(int x) {
        int tmp = y;
        y = 12;
        return x+tmp;
    }
}
  

File: src/frama-c-plugin/tests/jessie/pre_and_old.c



#pragma JessieIntegerModel(math)


int y;

//@ logic integer f{L}(integer n) = n+y;

/*@ ensures \result == \old(f(x));
  @ ensures \result == f{Old}(x);
  @ ensures \result == \at(f(x),Pre);
  @*/
int g(int x) {
  int tmp = y;
  y = 12;
  return x+tmp;
}
  

